mitiq/shadows/shadows.py

```diff
@@ -0,0 +1,140 @@
+    Union,
+    Sequence,
+    Tuple,
+    Dict,
+    Any,
+    cast,
+    List,
+)
+import cirq
+from functools import wraps
+import warnings
+import numpy as np
+
+from mitiq import Executor, Observable, QPROGRAM, QuantumResult
+from mitiq.shadows import *
+
+
+shadow_config = {"RShadow":False, # Use Robust Shadow Estimation or not
+                # Algorithm parameters (median of means) that users can tune (if RShadow = False, K1 and N1 are not used)
+                # One can choose to fill in K1, K2, N1, N2, or leave them as None if one wants to define error_rate and precision instead.
+                "K2": Optional[int],# Number of groups of "median of means" used for shadow estimation
+                "N2": Optional[int],# Number of shots per group of "median of means" used for shadow estimation
+                "K1": Optional[int],# Number of groups of "median of means" used for calibration in rshadow
+                "N1": Optional[int],# Number of shots per group of "median of means" used for calibration in rshadow
+                "error_rate": Optional[float], # epsilon
+                "precision": Optional[float], # 1 - delta
+                 }
+
+
+def execute_with_shadows(
+    circuit: cirq.Circuit,
+    executor: Union[Executor, Callable],
+    observables: Optional[List[Observable]] = None,
+    state_reconstruction: bool = False,
+    *,
+    # number of shots for shadow estimation
+    # w/o calibration (RShadow = False) then dim of the shadow is R_2
+    # w/ calibration (RShadow = True) then dim of the shadow is R_2, and dim of calibration is R_1.
+    # N1: Optional[int],
+    K1: Optional[int] = None,
+    R1: Optional[int] = None, # Number of Total Measurements for Classical shadow without calibration
+    # N2: Optional[int],
+    K2: Optional[int] = None,
+    R2: Optional[int] = None, # Number of Total Measurements for calibration
+
+    error_rate: Optional[float] = None, # epsilon
+    precision: Optional[float] = None, # 1 - delta
+    RShadow: Optional[bool] = False,
+    random_seed: Optional[int] = None,
+    max_batch_size: int = 100000000,
+) -> dict:
+    """
+    Executes a circuit with shadow measurements.
+    Args:  
+        circuit: The circuit to execute.
+        executor: The executor to use for running the circuit.
+        observables: The observables to measure. If None, the state will be reconstructed.
+        state_reconstruction: Whether to reconstruct the state or estimate the expectation value of the observables.
+        K1: Number of groups of "median of means" used for calibration in rshadow
+        R1: Total number of shots for calibration in rshadow
+        N1: Number of shots per group of "median of means" used for calibration in rshadow N1=R1/K1
+        K2: Number of groups of "median of means" used for shadow estimation
+        R2: Number of shots per group of "median of means" used for shadow estimation
+        N2: Total number of shots for shadow estimation N2=R2/K2
```

Comment:
Use more "pythonic" names. For example, `R1`-->`shots` or `estimation_shots` or `robust_shadow_shots`.

Reply:
Thanks, have renamed